src/main/java/it/geosolutions/geoserver/rest/GeoServerRESTPublisher.java

```diff
@@ -1520,6 +1522,215 @@ private boolean publishExternalCoverage(String workspace, String coveragestore,
     }
 
     // ==========================================================================
+    // === ARCGRID
+    // ==========================================================================
+
+    /**
+     * Upload and publish a ArcGrid image.
+     *
+     * @param workspace Workspace to use
+     * @param storeName The store name to be used or created.
+     * @param arcgrid The ArcGrid file.
+     * @return true if success.
+     * @throws FileNotFoundException if ArcGrid file does not exist.
+     */
+    public boolean publishArcGrid(String workspace, String storeName, File arcgrid)
+            throws FileNotFoundException {
+        return publishCoverage(workspace, storeName, CoverageStoreExtension.ARCGRID,
+                "image/arcgrid", arcgrid, ParameterConfigure.FIRST, (NameValuePair[]) null);
+    }
+
+    /**
+     * Upload and publish a ArcGrid image.
+     *
+     * @param workspace Workspace to use
+     * @param storeName Name of the coveragestore (if null the file name will be used)
+     * @param coverageName the name of the coverage (if null the file name will be used)
+     * @param arcgrid file to upload
+     * @return true if the operation completed successfully.
+     * @throws FileNotFoundException if file does not exists
+     * @throws IllegalArgumentException if workspace or arcgrid are null
+     */
+    public boolean publishArcGrid(final String workspace, final String storeName,
+                                  final String coverageName, final File arcgrid) throws FileNotFoundException,
+            IllegalArgumentException {
+        if (workspace == null || arcgrid == null)
+            throw new IllegalArgumentException("Unable to proceed, some arguments are null");
+
+        return publishCoverage(
+                workspace,
+                (storeName != null) ? storeName : FilenameUtils.getBaseName(arcgrid
+                        .getAbsolutePath()), CoverageStoreExtension.ARCGRID, "image/arcgrid",
+                arcgrid, ParameterConfigure.FIRST,
+                (coverageName != null) ? new NameValuePair[] { new NameValuePair("coverageName",
+                        coverageName) } : (NameValuePair[]) null);
+    }
+
+    /**
+     * Same as {@link #publishArcGrid(String, String, String, File, String, ProjectionPolicy, String, double[])} but without the last parameter
+     * (bbox). Kept here for backwards compatibility.
+     *
+     * @deprecated use the former method with bbox set to null.
+     */
```

Comment:
Being a new feature, I would avoid adding a deprecated method. 
You can directly rely on the method with the bbox param and remove this one.